### PR TITLE
Fixing most of CPP collation tests

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

**Comment by sven-oly** [comment]
This reduces failing tests from about 80K to less than 5K in CPP Collation tests.

File: executors/cpp/coll.cpp [hunk 1]
```diff
@@ -52,27 +55,59 @@ const string test_collator(json_object *json_in)  //
...
   } else {
-    locale_string = nullptr;
+    locale_string = "und";
   }
 
+  // Comparison type
+  json_object *compare_type_obj = json_object_object_get(json_in, "compare_type");
+  string compare_type_string = "";
+  if (compare_type_obj) {
+    compare_type_string = json_object_get_string(compare_type_obj);
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:77]: Yup, for future use.

File: executors/cpp/coll.cpp [hunk 2]
```diff
@@ -52,27 +55,59 @@ const string test_collator(json_object *json_in)  //
...
   const char *locale_string;
   if (locale_obj) {
     locale_string = json_object_get_string(locale_obj);
   } else {
-    locale_string = nullptr;
+    locale_string = "und";
   }
 
+  // Comparison type
+  json_object *compare_type_obj = json_object_object_get(json_in, "compare_type");
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:74]: Will do!

[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:74]: Added Issue https://github.com/unicode-org/conformance/issues/163 to record this.

File: executors/cpp/coll.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:77]: Added TODO to use this.

File: executors/cpp/coll.cpp [hunk 3]
```diff
@@ -86,36 +121,154 @@ const string test_collator(json_object *json_in)  //
...
+                           "error", json_object_new_string("error creating collator instance"));
+      no_error = false;
+      cout << "# Error in createInstance: " << label_string << " : " << test_result << endl;
+    }
 
-  // The json test output.
-  json_object *return_json = json_object_new_object();
-  json_object_object_add(return_json, "label", label_obj);
+    if (strength_obj) {
+      // !!!
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:182]: Removed this, and also extraneous "cout" statements. Tnx!

**Comment by sven-oly** [comment]
PTAL.

**Review by sffc — APPROVED** [review]

**Pull request merged**